docs/guide/web-simulation.md

```diff
@@ -184,22 +182,92 @@ HTTP request handlers:
+In order to make the web simulation work properly from the outside world, you have two possibilities:
+1. Configure your web server, `session_server.py` and `simulation_server.py` to perform port rewrite.
+Requests arriving to the web server will be redirected to the local network, e.g., `wss://webserver.com/2000/client` &rarr; `ws://webserver.com:2000/client` and `https://webserver.com/2000/monitor` &rarr; `http://webserver.com:2000/monitor`.
+You just need to have port 443 open in your local firewall for incoming connections.
+Such a setup is compatible with any client firewall allowing outgoing connections on port 443 to your server (which is very standard).
+Moreover, it has the advantage of allowing you to run the `session_server.py`, `simulation_server.py` and Webots without SSL encryption while still proving a SSL interface to the clients via your web server.
+2. Open all the ports used by the `session_server.py`, `simulation_server.py` and Webots.
+On simple networks, this can be done by modifying the NAT settings of the router.
+The disadvantage of this method is that local firewalls may block the connections on non-standard ports, and thus some clients may not be able to use the web simulation if they cannot change their firewall rules.
+Another disadvantage is that you must run the `session_server.py`, `simulation_server.py` and Webots in SSL mode if you want to provide a SSL connection to the simulation server, so that it can be executed from a `https` web page.
+
+#### Port Rewrite
+
+Port rewrite is useful for providing an SSL interface using the standard 443 port for both WebSockets `wss://` and HTTP `https://` requests.
+It can be tested on a local host by setting up [XAMPP to use SSL](https://gist.github.com/nguyenanhtu/33aa7ffb6c36fdc110ea8624eeb51e69).
+Port rewrite should be configured in the `session_server.py` and `simulation_server.py` by simply setting the `portRewrite` option to `true`.
+In the `session_server.py` configuration, the `simulationServers` should be listed using the outside URL: `hostname/2000` instead of `hostname:2000`.
+Your web server should be configured to redirect `http` traffic to `https` and to rewrite ports in URLs for both `https` and `wss`.
+With the Apache web server, this can be achieved by adding the following rules in your `httpd.conf` file:
+
+```
+LoadModule proxy_module modules/mod_proxy.so
+LoadModule proxy_http_module modules/mod_proxy_http.so
+LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
+
+<VirtualHost *:80>
+  ServerName webserver.com
+  ServerAlias www.webserver.com
+
+  [ ... ]
+
+  RewriteEngine on
+
+  RewriteCond %{SERVER_NAME} =%{SERVER_NAME} [OR]
+  RewriteCond %{SERVER_NAME} =www.%{SERVER_NAME}
+  RewriteRule ^ https://%{SERVER_NAME}%{REQUEST_URI} [END,NE,R=permanent]
+</VirtualHost>
+
+<VirtualHost *:443>
+  ServerName webserver.com
+  ServerAlias www.webserver.com
+
+  [ ... ]
+
+  RewriteEngine on
+
+  RewriteCond %{SERVER_NAME} =www.webserver.com
```

Comment:
I would add a comment describing that this rule redirects to the main domain (without www) and that should be skipped for localhost